server/src/main/java/org/elasticsearch/cluster/metadata/AliasValidator.java

```diff
@@ -150,4 +160,58 @@ private static void validateAliasFilter(XContentParser parser, QueryShardContext
         QueryBuilder queryBuilder = Rewriteable.rewrite(parseInnerQueryBuilder, queryShardContext, true);
         queryBuilder.toFilter(queryShardContext);
     }
+
+    /**
+     * throws exception if specified alias is attempting to be write_only while this alias points to another index
+     *
+     * @param aliasName the alias in question
+     * @param metaData the Cluster metadata
+     * @throws IllegalArgumentException if the alias cannot be write_only
+     */
+    public void validateAliasWriteOnly(String aliasName, boolean isWriteIndex, MetaData metaData) {
+        if (isWriteIndex) {
+            AliasOrIndex aliasOrIndex = metaData.getAliasAndIndexLookup().get(aliasName);
+            if (aliasOrIndex != null && aliasOrIndex.isAlias()) {
+                AliasOrIndex.Alias alias = (AliasOrIndex.Alias) aliasOrIndex;
+                List<IndexMetaData> writeIndices = alias.getWriteIndices();
+                if (writeIndices.size() >= 1) {
+                    throw new IllegalArgumentException("Alias [" + aliasName +
+                        "] already has a write index set [" + writeIndices.get(0).getIndex().getName() + "]");
+                }
+            }
+        }
+    }
+
+    public void validateAliasWriteOnly(String checkAlias, String checkIndex, boolean writeIndex,
+                                       ImmutableOpenMap.Builder<String, IndexMetaData> indices) {
+        if (writeIndex) {
+            Index existing = getExistingWriteIndex(checkAlias, indices);
+            if (existing != null && existing.getName().equals(checkIndex) == false) {
+                throw new IllegalArgumentException("Alias [" + checkAlias +
+                    "] already has a write index set [" + existing.getName() + "]");
+            }
+        }
+    }
+
+    public static boolean validAliasWriteOnly(String aliasName, boolean writeIndex,
+                                              ImmutableOpenMap.Builder<String, IndexMetaData> indices) {
+        return writeIndex && getExistingWriteIndex(aliasName, indices) == null;
+    }
+
+    public static boolean validAliasWriteOnly(String aliasName, boolean writeIndex, ImmutableOpenMap<String, IndexMetaData> indices) {
```

Comment:
can we get another naming convention? we use validate for exceptions and valid is just too close. Also, unit tests please.